src/dmd/parse.d

```diff
@@ -6029,19 +6029,56 @@ final class Parser(AST) : Lexer
         switch (token.value)
         {
         case TOK.leftCurly:
-            /* Scan ahead to see if it is a struct initializer or
-             * a function literal.
-             * If it contains a ';', it is a function literal.
-             * Treat { } as a struct initializer.
+            /* Scan ahead to discern between struct initializer and
+             * parameterless function literal.
+             *
+             * We'll scan the topmost curly bracket level for statement-related
+             * tokens, thereby ruling out a struct initializer.  (A struct
+             * initializer which itself containins function literals may have
+             * statements at nested curly bracket levels.)
+             *
+             * It's important that this functional literal check not be
```

Comment:
It's important for this function literal check to be

Reply:
Not sure I understand.  You'd like to change the polarity of the statement, from "the check should not be pedantic" to "the check should be pedantic"?

Responding to your other comment, this matter was indeed not obvious to me so seems worthy of source code comment.  If the author _intended_ a function literal (implies heuristics as employed here), then even when that literal has a syntax error we want to parse it as a function and emit understandable errors.

Reply:
Ah I think you just meant "functional" --> "function" here.   Done.

I'll move the other matter to your corresponding comment.

Reply:
Oh I was just trying to point out that "It's important that XXX not be" is missing a verb or in other words "not be" sounded really strange.